frontend/server/src/Controllers/User.php

```diff
@@ -4328,6 +4328,14 @@ public static function isNonUserIdentity(
      * @omegaup-request-param string $third_party_login
      */
     public static function getLoginDetailsForTypeScript(\OmegaUp\Request $r) {
+        if (
+            !is_null(
+                \OmegaUp\Controllers\Session::getCurrentSession()['identity']
+            )
+        ) {
+            header('Location: /');
+            throw new \OmegaUp\Exceptions\ExitException();
+        }
```

Comment:
```suggestion
        try {
            $r->ensureIdentity();
            // If the user has already logged in, redirect them to the home page.
            header('Location: /');
            throw new \OmegaUp\Exceptions\ExitException();
        } catch (\OmegaUp\Exceptions\UnauthorizedException $e) {
            // Do nothing.
        }
```

we don't use `getCurrentSession()` in a lot of places. This pattern is a lot more used.